pkg/sql/schemaexpr/column.go

```diff
@@ -234,6 +234,27 @@ func replaceVars(
 			return true, expr, nil
 		}
 
+		// Check if a column is being referenced from another table. It may be the
+		// case where the column being referenced has the same name as a column
+		// in our current table.
+		if c.TableName != nil {
+			// If NumParts > 1, a database or schema name was provided.
+			// To disallow cross database / schema column references, we don't
+			// allow database and schema identifiers with the column name.
+			if c.TableName.NumParts > 1 {
+				return false, nil,
+					pgerror.Newf(pgcode.InvalidColumnReference,
+						"cannot reference columns with a specified database or schema")
```

Comment:
I think something like "cannot qualify columns in expression with database or schema" would be clearer.